### DISCUSSION TITLE
Excited for this

### DISCUSSION THREAD

**Discussion by olimorris** [stevearc/overseer.nvim#1] [General]
I've been writing a few Neotest adapters recently and one of the things that stood out to me was how nice it would be to trigger a full test suite run in the background and see an icon in my statusline to reflect its outcome. This looks uber promising and can't wait!

**Comment by stevearc** [comment]
Glad you're excited! I've still got a ways to go before official release (mostly writing up documentation), but hopefully it'll be ready soon!

**Reply by olimorris** [reply]
If it's useful for me to do some testing, let me know

**Reply by olimorris** [reply]
Finally (!!) got around to testing this. What a joy. Love the API and that I have to do barely any config to hook into Neotest.

I've cobbled together an integration for [feline.nvim](https://github.com/feline-nvim/feline.nvim) which is almost working so I shall share that when it's 100%.

Being able to run the whole test suite and see a beautiful icon in the statusline is a joy. Or I pop open Overseer if there is a failure. Bravo 👏🏼.

Would love to know if you're doing anything differently with Neotest.

**Reply by stevearc** [reply]
Thanks for the kind words! Would love to include the feline.nvim integration when it's completed.

The main thing that I use the overseer neotest integration for is rerunning tests on save. If I have a test that's failing, I'll trigger a run and then take the resulting overseer task and run the "watch" action on it (which just adds the `restart_on_save` component). Now the test will re-run every time I make a change to a file in the project. I also modify the `on_complete_notify` component to set `on_change = true`, so that I only see notifications when the test goes from failing to passing or vice versa.

**Reply by olimorris** [reply]
Nice!!! That is a really sick workflow. I couldn't see it in your dotfiles but will totally steal that.

**Comment by stevearc** [comment]
Beta is released and [Neotest support](https://github.com/stevearc/overseer.nvim#neotest) is working! (including [task restarting](https://github.com/stevearc/overseer.nvim/commit/3e8763e997f295e908f2704fe025da97df645ba4)!)
